### PR TITLE
add reading ability and reloading netcdf/zarr stores

### PR DESCRIPTION
**Pull request by dallasfoster**
# Earth2Studio Pull Request

## Description
- Added the ability to reload zarr and netcdf backends
- Added the ability to read from an IOBackend
- NetCDF reads are now mode='r+' instead of 'w'.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: earth2studio/io/netcdf4.py [hunk 1]
```diff
@@ -51,13 +51,33 @@ def __init__(
         # Set persist to false if diskless is false
         self.root = Dataset(
             file_name,
-            "w",
+            "r+",
```
Comment (NickGeneva) [inline earth2studio/io/netcdf4.py:54]: Will this fix the issue with running an example twice and it erroring because the file already exists?

Comment (dallasfoster) [reply earth2studio/io/netcdf4.py:54]: Yes, this should fix that issue. Both with zarr and netcdf.

File: earth2studio/io/netcdf4.py [hunk 2]
```diff
@@ -51,13 +51,33 @@ def __init__(
...
             format="NETCDF4",
             diskless=diskless,
             persist=persist if diskless else False,
         )
 
         self.coords: CoordSystem = OrderedDict({})
+        for dim in self.root.dimensions:
+            print(dim)
```
Comment (NickGeneva) [inline earth2studio/io/netcdf4.py]: Lot of debug print statements

**Review by NickGeneva — APPROVED** [review]

**Pull request merged**